src/hibernate-resume/hibernate-resume.c

```diff
@@ -29,17 +32,39 @@ int main(int argc, char *argv[]) {
                 return EXIT_SUCCESS;
 
         device = argv[1];
+        offset = getenv("SYSTEMD_HIBERNATE_RESUME_OFFSET");
 
         if (stat(device, &st) < 0) {
                 log_error_errno(errno, "Failed to stat '%s': %m", device);
                 return EXIT_FAILURE;
         }
 
         if (!S_ISBLK(st.st_mode)) {
-                log_error("Resume device '%s' is not a block device.", device);
+                log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Resume device '%s' is not a block device.", device);
                 return EXIT_FAILURE;
         }
 
+        if (offset) {
+                if (access("/sys/power/resume_offset", W_OK) < 0) {
```

Comment:
this has a TTOCTTOU feeling to it: drop the explicit initial access() check, and just write immediately to the sysattr. We'll get ENOENT from that much the same and can handle things exactly the same way, but with shorter code and without TTOCTTOU.

(i know the current sleep code does that too, but it really should be fixed there too)